client/data/emails/use-remove-titan-mailbox-mutation.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { useCallback } from 'react';
+import { useMutation, useQueryClient } from 'react-query';
+
+/**
+ * Internal dependencies
+ */
+import { queryKeyPrefix } from './use-emails-query';
+import wp from 'calypso/lib/wp';
+
+const noop = () => {};
+
+/**
+ * Deletes a mailbox from a Professional Email (Titan) account
+ *
+ * @param {string} domainName The domain name of the mailbox
+ * @param {string} mailboxName The mailbox name
+ * @param {object} mutationOptions Mutation options passed on to `useMutation`
+ * @returns {{ data, error, isLoading: boolean, removeTitanMailbox: Function, ...}} Returns various parameters piped from `useMutation`
+ */
+export function useRemoveTitanMailboxMutation( domainName, mailboxName, mutationOptions = null ) {
+	mutationOptions = mutationOptions ?? {};
+
+	const suppliedOnSettled = mutationOptions.onSettled ?? noop;
+
+	const queryClient = useQueryClient();
+
+	mutationOptions.onSettled = ( data, error, variables, context ) => {
+		suppliedOnSettled?.( data, error, variables, context );
+
+		queryClient.invalidateQueries( {
+			predicate: ( { queryKey: [ prefix, , domain ] = [] } ) =>
+				prefix === queryKeyPrefix && domain === domainName,
+		} );
```

Comment:
As far I can see, we are no longer taking the site identifier in consideration when invalidating data. Wouldn't that be a problem? Why do we need to define a custom predicate? Can't we reuse our cache key like this instead?

```suggestion
		queryClient.invalidateQueries( getCacheKey( siteId, domainName ) );
```

Reply:
We don't need to pass in the siteId just to invalidate the cache. React-query is flexible enough to work with a predicate.

Reply:
I understand we can pass a predicate. The thing that I don't is that we use a cache key when are fetching data that is different from when we are mutating that same data.

Reply:
It's more accurately called a `queryKey` not a cache key. We can supply a portion, or prefix of the key to get the runtime to invalidate all matching queries.